### DISCUSSION TITLE
Need a new search node: 1 big one or multiple smaller ones?

### DISCUSSION THREAD

**Discussion by royston-boot** [Security-Onion-Solutions/securityonion#6033] [Unsupported Versions] [answered]
Hi all

I have a 2.3.80 distributed architecture with a sensor node on a 10Gb/s Internet link usually running at about 800Mb/s. Packet loss is acceptable. However the search node is massively underpowered and running at a CPU load of 8 to 12 across it's 6 cores (gulp!).  I am likely to add another dedicated sensor node for our data centre which will add perhaps another 200Mb/s of traffic.  

I have some budget to improve the search situation.

Do you think it is better to add multiple smaller search nodes or a single more powerful node?
In a multi search node setup, will the manager allocate work intelligently based on the available resources on the search nodes even if they are quite different, CPU/RAM wise?

I can afford ONE dual-AMD EPYC 8 core (16C total), 256GB DDR4 ram, m.2 sata SSD for boot/system and 16TB of 6Gb Entry SATA SSD OR I could buy TWO single Xeon 6 Core, 128GB ram, m.2 sata SSD for boot/system and 8TB of 6Gb entry level SATA SSD. I *might* just be able to get three of the Xeon 6C servers.
(Lenovo SR645 and SR250 FWIW).

Option 1 would give me 2 nodes: one fast and one slow and option 2 would give me 3 (or 4) similarly spec'd search nodes.
I would appreciate your thoughts.

cheers
Royston

**Comment by dougburks** [comment, marked as answer]
> In a multi search node setup, will the manager allocate work intelligently based on the available resources on the search nodes even if they are quite different, CPU/RAM wise?

Each of the search nodes will try to consume new logs from the redis queue on the manager as fast as they can.

> Do you think it is better to add multiple smaller search nodes or a single more powerful node?

If the total hardware specs of each option are roughly the same, then you may get roughly the same performance with either option. More nodes might get slightly better performance due to parallelization, but there are lots of variables here.

**Reply by royston-boot** [reply]
Thanks Doug. Think we'll go with the bigger box - less cabling and management hassles.
